pkg/storage/segments/store.go

```diff
@@ -311,21 +311,26 @@ func (s *segmentStore) Repair(ctx context.Context, path storj.Path, lostPieces [
 
 	// get the nodes list that needs to be excluded
 	var excludeNodeIDs []dht.NodeID
+	healthyNodes := make([]*pb.Node, len(originalNodes))
 
 	// count the number of nil nodes thats needs to be repaired
 	totalNilNodes := 0
+
+	OUTER:
 	for j, v := range originalNodes {
 		if v != nil {
 			excludeNodeIDs = append(excludeNodeIDs, node.IDFromString(v.GetId()))
-		} else {
-			totalNilNodes++
-		}
 
-		//remove all lost pieces from the list to have only healthy pieces
-		for i := range lostPieces {
-			if j == int(lostPieces[i]) {
-				totalNilNodes++
+			// If node index exists in lostPieces, skip adding it to healthyNodes
+			for i := range lostPieces {
+				if j == int(lostPieces[i]) {
+					totalNilNodes++
+					continue OUTER
```

Comment:
Let's avoid using this OUTER label, by extracting a hepler `contains` function. Then here we would have a much clearer code like:
```go
if contains(lostPieces, j) {
    totalNilNodes++
} else {
    healthyNodes[j] = v
}
```